Documentation/user-guides/configuring-cluster-monitoring.md

```diff
@@ -1,137 +1,858 @@
+  # The client cert in the Prometheus container to use for the targets.
+  cert: v1.SecretKeySelector
+  # Disable target certificate validation.
+  insecureSkipVerify: <bool>
+  # The client key in the Prometheus container to use for the targets.
+  key: v1.SecretKeySelector
+  # Used to verify the hostname for the targets.
+  serverName: <string>
+```
+
+### RemoteWriteSpec
+
+Holds configuration for remote write
+
+```yaml
+remoteWrite:
+  # The URL of the endpoint to send samples to.
+  url: <string>
+  # The name of the remote write queue, must be unique if specified. The
+  # name is used in metrics and logging in order to differentiate queues.
+  # Only valid in Prometheus versions 2.15.0 and newer.
+  name: <string>
+  # Timeout for requests to the remote write endpoint.
+  remoteTimeout: <string>
+  # Custom HTTP headers to be sent along with each remote write request.
+  # Be aware that headers that are set by Prometheus itself can't be overwritten.
+  # Only valid in Prometheus versions 2.25.0 and newer.
+  headers: <map[string]string>
+  # The list of remote write relabel configurations.
+  writeRelabelConfigs: RelabelConfig
+  # BasicAuth for the URL.
+  basicAuth: BasicAuth
+  # Bearer token for remote write.
+  bearerTokenFile: <string>
+  # TLS Config to use for remote write.
+  tlsConfig: SafeTLSConfig
+  # Optional ProxyURL
+  proxyUrl: <string>
+  # QueueConfig allows tuning of the remote write queue parameters.
+  queueConfig: QueueConfig
+  # MetadataConfig configures the sending of series metadata to remote storage.
+  metadataConfig: MetadataConfig
+```
+
+
+#### BasicAuth
+
+```yaml
+basicAuth:
+  # The secret in the service monitor namespace that contains the username
+  # for authentication.
+  username: v1.SecretKeySelector
+  # The secret in the service monitor namespace that contains the password
+  # for authentication.
+  password: v1.SecretKeySelector
+```
+
+#### MetadataConfig
 
 ```yaml
-# baseImage is the container image repository that will be used to deploy the node-exporter pods
-baseImage: <string>
+metadataConfig:
+  # Whether metric metadata is sent to remote storage or not.
+  send: <bool>
+  # How frequently metric metadata is sent to remote storage.
+  sendInterval: <string>
 ```
-### KubeStateMetricsConfig
 
-Use KubeStateMetricsConfig to configure parameters for deployment of the `kube-state-metrics` components.
+#### QueueConfig
 
 ```yaml
-# baseImage is the container image repository that will be used to deploy the kube-state-metrics pods
-baseImage: <string>
-addonResizerBaseImage: <string>
+queueConfig:
+  # BatchSendDeadline is the maximum time a sample will wait in buffer.
+  batchSendDeadline: <string>
+  # Capacity is the number of samples to buffer per shard before we start dropping them.
+  capacity: <integer>
+  # MaxBackoff is the maximum retry delay.
+  maxBackoff: <string>
+  # MaxRetries is the maximum number of times to retry a batch on recoverable errors.
+  maxRetries: <integer>
```

Comment:
from https://github.com/openshift/cluster-monitoring-operator/blob/release-4.11/vendor/github.com/prometheus/prometheus/config/config.go#L834-L855
there is not `maxRetries`